### PR TITLE
Cleanup dates.py docstrings.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/dates.py [hunk 1]
```diff
@@ -568,7 +568,8 @@ def drange(dstart, dend, delta):
 
 class DateFormatter(ticker.Formatter):
     """
-    Format a tick (in seconds since the epoch) with a `strftime` format string.
+    Format a tick (in seconds since the epoch) with a
+    `~datetime.datetime.strftime` format string.
```
Comment (jklymak) [inline lib/matplotlib/dates.py]: This isn't even correct is it?  Its decimal days since the epoch.

File: lib/matplotlib/dates.py [hunk 2]
```diff
@@ -1223,57 +1220,61 @@ def autoscale(self):
...
+            locator = AutoDateLocator()
+            locator.intervald[HOURLY] = [3]  # only show every 3 hours
+    """
+
+    def __init__(self, tz=None, minticks=5, maxticks=None,
+                 interval_multiples=True):
+        """
+        Parameters
+        ----------
+        tz : `tzinfo`
```
Comment (jklymak) [inline lib/matplotlib/dates.py]: Maybe: "Specify the timezone for the ticks".

File: lib/matplotlib/dates.py [hunk 3]
```diff
@@ -1223,57 +1220,61 @@ def autoscale(self):
...
+        minticks : int
+            The minimum number of ticks desired; controls whether ticks occur
+            yearly, monthly, etc.
+        maxticks : int
+            The maximum number of ticks desired; controls the interval between
+            ticks (ticking every other, every 3, etc.).  For fine-grained
+            control, this can be a dictionary mapping individual rrule
+            frequency constants (YEARLY, MONTHLY, etc.) to their own maximum
+            number of ticks.  This can be used to keep the number of ticks
```
Comment (jklymak) [inline lib/matplotlib/dates.py]: Can you reference `intervald` above?

File: lib/matplotlib/dates.py [hunk 4]
```diff
@@ -1223,57 +1220,61 @@ def autoscale(self):
...
+            appropriate to the format chosen in `AutoDateFormatter`. Any
+            frequency not specified in this dictionary is given a default
+            value.
+        interval_multiples : bool
```
Comment (jklymak) [inline lib/matplotlib/dates.py]: State default is True

File: lib/matplotlib/dates.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/dates.py]: indeed...

File: lib/matplotlib/dates.py (hunk 3 above)
Comment (anntzer) [reply lib/matplotlib/dates.py]: tbh it's not clear to me how intervald interacts with minticks, maxticks, and interval_multiples, so I'll pass on that...

File: lib/matplotlib/dates.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/dates.py]: done

File: lib/matplotlib/dates.py (hunk 4 above)
Comment (anntzer) [reply lib/matplotlib/dates.py]: done

File: lib/matplotlib/dates.py (hunk 2 above)
Comment (QuLogic) [reply lib/matplotlib/dates.py]: Is it?

Comment (anntzer) [reply lib/matplotlib/dates.py]: oops, had only fixed one of the occurrences, now the second one is good too.

**Pull request merged**